xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-ui/src/main/resources/XWiki/Notifications/Code/PageNotificationEventDescriptorClass.xml

```diff
@@ -0,0 +1,194 @@
+    <comment/>
+    <minorEdit>false</minorEdit>
+    <syntaxId>xwiki/2.1</syntaxId>
+    <hidden>true</hidden>
+    <content/>
+    <class>
+        <name>XWiki.Notifications.Code.PageNotificationEventDescriptorClass</name>
+        <customClass/>
+        <customMapping/>
+        <defaultViewSheet/>
+        <defaultEditSheet/>
+        <defaultWeb/>
+        <nameField/>
+        <validationScript/>
+        <applicationName>
+            <customDisplay/>
+            <defaultValue/>
+            <disabled>0</disabled>
+            <name>applicationName</name>
+            <number>1</number>
+            <picker>0</picker>
+            <prettyName>Application</prettyName>
+            <size>64</size>
+            <unmodifiable>0</unmodifiable>
+            <validationMessage/>
+            <validationRegExp/>
+            <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+        </applicationName>
+        <eventType>
+            <customDisplay/>
+            <defaultValue/>
+            <disabled>0</disabled>
+            <name>eventType</name>
+            <number>2</number>
+            <picker>0</picker>
+            <prettyName>Event type</prettyName>
+            <size>64</size>
+            <unmodifiable>0</unmodifiable>
+            <validationMessage/>
+            <validationRegExp/>
+            <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+        </eventType>
+        <eventPrettyName>
+            <customDisplay/>
+            <defaultValue/>
+            <disabled>0</disabled>
+            <name>eventPrettyName</name>
+            <number>3</number>
+            <picker>0</picker>
+            <prettyName>Event name</prettyName>
+            <size>64</size>
+            <unmodifiable>0</unmodifiable>
+            <validationMessage/>
+            <validationRegExp/>
+            <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+        </eventPrettyName>
+        <eventIcon>
+            <customDisplay/>
+            <defaultValue/>
+            <disabled>0</disabled>
+            <name>eventIcon</name>
+            <number>4</number>
+            <picker>0</picker>
+            <prettyName>Event icon</prettyName>
+            <size>64</size>
+            <unmodifiable>0</unmodifiable>
+            <validationMessage/>
+            <validationRegExp/>
+            <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+        </eventIcon>
+        <listenTo>
+            <customDisplay/>
+            <defaultValue/>
+            <disabled>0</disabled>
+            <name>listenTo</name>
+            <number>5</number>
+            <picker>0</picker>
+            <prettyName>Listen to …</prettyName>
+            <size>64</size>
+            <unmodifiable>0</unmodifiable>
+            <validationMessage/>
+            <validationRegExp/>
+            <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
```

Comment:
> If you don't use relational storage, you are limited by the size of the field that stores the values of the list (usually a string).

No it's not a string. See ListClass#newProperty() implementation.

>  So ok, if you are sure that a static list using the input displayer will be stored in the database using the relational storage, then yes it's the correct approach.

It's not related to the input displayer but to multiselect option. If you plan to use the default displayer and not do your own thing you should also make sure to select the right separators for it to not conflict with characters that could be part of your values.

Here are a few examples:
* XWiki.ConfigurableClass_propertiesToShow
* many properties in ExtensionCode.Extension*Class classes
* Watchlist is using this too even if for some weird reason it's using addDBListField with no class or sql request which does not make much sense but have the same effect (everything is done at ListClass level anyway)